regression/cbmc/Sideeffects7/main.c

```diff
@@ -0,0 +1,15 @@
+int main()
+{
+  struct
+  {
+    int a : 2;
+  } b;
+  int c;
+  short d;
+  // this is to make sure the fix for the below assignment does not introduce a
+  // new bug
+  d |= b.a = c >= 0;
+
+  // we previously weren't able to handle the below
+  short a = (a ^= 0) || __assert_fail("", 2, __PRETTY_FUNCTION__);
+}
```

Comment:
I would replace the architecture-specific `__assert_fail` by some other side effect, say `x++` or the like.

Reply:
Done (I've added a call to a newly added function, which in turn calls `__CPROVER_assert`).